extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -35,8 +34,8 @@ public ThreadContextSnapshot currentContext(Map<String, String> map) {
             return NOOP_SNAPSHOT;
```

Comment:
@manovotn I wonder if this is wrong: supposedly we may want to capture the fact that there's no request context, so that if we ever restore that context somewhere where this is another request context we would temporarily deactivate it?

Reply:
You are right in theory but the practice is that we are doing the propagation and we are responsible for cleaning up afterwards (deactivate context). Following that principle, there cannot really be an active req. context on the "new" thread. Hence we have this `NOOP` snapshot here.

Reply:
That's not true: 

```java
Runnable hasNoContext = threadContext.withContextCapture(() -> { /* no request context */});

requestContext.activate();

hasNoContext.run(); // has a request context because capture/restore is NOOP

requestContext.deactivate();
```

Now, this is academic at this point because it requires an action to be created in a thread with no request context active (perhaps the `@OnInit` beans?), receiving a notification from a request thread, which will be rare, but not impossible.

Reply:
Doing that just breaks everything, let me show you a vice-versa example of capturing a valid context a then shutting it down manually. The logic is equally skewed and it doesn't make sense to do that:

```
Runnable hasContext = threadContext.withContextCapture(() -> { /* active req. context exists here */});

requestContext.ternminate(); // destroys the whole map, even for propagated context that didnt run yet

hasContext.run(); // now this expects propagated context but there will be none now
```
E.g. we are now not capturing instances but at certain point in time (== when snapshot is created) we have to capture whatever is in the thread local Map field. If there is no active req. context, then there is no map and nothing can be put into the snapshot - simple as that.

Or maybe I still don't understand what you mean?

Reply:
Well, I'm only simulating request activation to show my point. `terminate` is done by the container, so that's another story, but the point is that if you have a contextual "thing" created outside a request, and it ends up being invoked withing a request it will inherit the request, which is not the expected outcome.

Reply:
Aah I think i finally undestand what you mean - that NOOP doesn't suspend&restore which theoretically might cause problems if the contextualized action runs on the same thread. Well, you are right there. Though from user perspective if you have an action that is able to run w/o req. context, having it won't making a difference. But we can definitely fix that. Lemme add a commit.

Reply:
@FroMage added a commit that should resolve that unless you meant something else ;)

Reply:
> Though from user perspective if you have an action that is able to run w/o req. context, having it won't making a difference

Well, maybe it would behave differently accidentally if it had a request context?

Reply:
But yeah, thanks.